qemu/tests/cfg/cpu_topology_test.cfg

```diff
@@ -0,0 +1,12 @@
+- cpu_topology_test:
+    no smp2
+    only Linux
+    type = cpu_topology_test
+    start_vm = no
+    check_sockets_cmd = 'cat /proc/cpuinfo | grep "physical id" | sort | uniq |wc -l '
```

Comment:
@vivianQizhu  Hi ，The  three  cmdline not  support for aarch64. 
You can use "/sys/devices“ interface. It's more general and should apply to all architectures, possibly including ppc. 
 eg：check_sockets_cmd  = ‘ls /sys/devices/system/node | grep node | wc -l’

Reply:
Thanks @Kuhn-Chen 
@nanliu-r Would you help check if we can update our test cases with these commands?

Reply:
Yes，of course！

Reply:
> Thanks @Kuhn-Chen
> @nanliu-r Would you help check if we can update our test cases with these commands?

I can't get correct sockets with cmd ‘ls /sys/devices/system/node | grep node | wc -l’ , can you help check this ? and we can check topology with " /sys/devices/system/cpu/cpu*/topology" to check echo cpu but seems more complicated.

Reply:
@PaulYuuu Would you give a hand on this, how do we test this on aarch64 usually? Thanks.

Reply:
Hello guys,
1. `/sys/devices/system/node` means NUMA node, not CPU sockets
2. `ls /sys/devices/system/cpu/ | grep -P "cpu\d+" | wc -l` shows the count of CPU. I agree that `/sys/devices/system/cpu/cpu*/topology` is a general one but more complicated as @nanliu-r said, because we need to check per CPU.

In my opinion, `lscpu` is the best choice for all architectures, `cpuinfo` shows all CPU details, it's too long.

@vivianQizhu I attach both `lscpu` and `cpuinfo` of 'x86', 'ppc', and 'aarch64', you can refer to them and pick out the best you think.

x86:
```
Architecture:          x86_64
CPU op-mode(s):        32-bit, 64-bit
Byte Order:            Little Endian
CPU(s):                8
On-line CPU(s) list:   0-7
Thread(s) per core:    2
Core(s) per socket:    4
Socket(s):             1
NUMA node(s):          1
Vendor ID:             GenuineIntel
CPU family:            6
Model:                 142

processor	: 7
vendor_id	: GenuineIntel
cpu family	: 6
model		: 142
model name	: Intel(R) Core(TM) i7-8650U CPU @ 1.90GHz
stepping	: 10
microcode	: 0xca
cpu MHz		: 3420.062
cache size	: 8192 KB
physical id	: 0
siblings	: 8
core id		: 3
cpu cores	: 4
apicid		: 7
initial apicid	: 7
```

ppc:
```
Architecture:        ppc64le
Byte Order:          Little Endian
CPU(s):              160
On-line CPU(s) list: 0-159
Thread(s) per core:  4
Core(s) per socket:  20
Socket(s):           2
NUMA node(s):        2
Model:               2.2 (pvr 004e 1202)
Model name:          POWER9, altivec supported

processor	: 159
cpu		: POWER9, altivec supported
clock		: 3600.000000MHz
revision	: 2.2 (pvr 004e 1202)
```

aarch64:
```
Architecture:        aarch64
Byte Order:          Little Endian
CPU(s):              8
On-line CPU(s) list: 0-7
Thread(s) per core:  1
Core(s) per socket:  2
Socket(s):           4
NUMA node(s):        1
Vendor ID:           APM

processor	: 7
BogoMIPS	: 100.00
Features	: fp asimd evtstrm cpuid
CPU implementer	: 0x50
CPU architecture: 8
CPU variant	: 0x0
CPU part	: 0x000
CPU revision	: 1
```

Reply:
BTW, x86 introduce another property of CPU topology named `die`, it might be a checkpoint.
In addition, `provider.cpu_utils.check_guest_cpu_topology` may be suitable for this scenario except `die`.

Reply:
> BTW, x86 introduce another property of CPU topology named `die`, it might be a checkpoint.
> In addition, `provider.cpu_utils.check_guest_cpu_topology` may be suitable for this scenario except `die`.

Thanks, we have another case designed for property `die` since we are going to add that to gating.

Reply:
> BTW, x86 introduce another property of CPU topology named `die`, it might be a checkpoint.
> In addition, `provider.cpu_utils.check_guest_cpu_topology` may be suitable for this scenario except `die`.

Yes, `provider.cpu_utils.check_guest_cpu_topology` fit this test very well and it covers Windows as well, I will update it accordingly. Thanks a lot @PaulYuuu

Reply:
> 1. `/sys/devices/system/node` means NUMA node, not CPU sockets

Yes，the value of node is not always equal to that of socket. 

In the code of the lscpu tool,  in the read_topology() , "nsockets = desc->ncpus / nthreads / ncores;".
https://github.com/karelzak/util-linux/blob/master/sys-utils/lscpu.c

Reply:
> In my opinion, `lscpu` is the best choice for all architectures, `cpuinfo` shows all CPU details, it's too long.

I think it's OK to use lscpu because it's also using  kernel interface "/sys/devices/system/".